server/src/main/java/org/apache/druid/client/indexing/HttpIndexingServiceClient.java

```diff
@@ -194,26 +194,39 @@ public String cancelTask(String taskId)
 
   @Override
   public int getTotalWorkerCapacity()
+  {
+    return getWorkers().stream().mapToInt(workerInfo -> workerInfo.getWorker().getCapacity()).sum();
+  }
+
+  @Override
+  public int getTotalWorkerCapacityWithAutoScale()
   {
     try {
       final StringFullResponseHolder response = druidLeaderClient.go(
-          druidLeaderClient.makeRequest(HttpMethod.GET, "/druid/indexer/v1/workers")
+          druidLeaderClient.makeRequest(HttpMethod.GET, "/druid/indexer/v1/autoScaleConfig")
                            .setHeader("Content-Type", MediaType.APPLICATION_JSON)
       );
-
       if (!response.getStatus().equals(HttpResponseStatus.OK)) {
         throw new ISE(
-            "Error while getting available cluster capacity. status[%s] content[%s]",
+            "Error while getting worker info. status[%s] content[%s]",
             response.getStatus(),
             response.getContent()
         );
       }
-      final Collection<IndexingWorkerInfo> workers = jsonMapper.readValue(
+      final IndexingAutoScaleConfigInfo indexingAutoScaleConfigInfo = jsonMapper.readValue(
           response.getContent(),
-          new TypeReference<Collection<IndexingWorkerInfo>>() {}
+          new TypeReference<IndexingAutoScaleConfigInfo>() {}
       );
-
-      return workers.stream().mapToInt(workerInfo -> workerInfo.getWorker().getCapacity()).sum();
+      Collection<IndexingWorkerInfo> workers = getWorkers();
+      int capacityPerWorker;
+      if (workers != null && !workers.isEmpty()) {
+        capacityPerWorker = workers.stream().findFirst().get().getWorker().getCapacity();
```

Comment:
This assumes all nodes have the same capacity, but if you set up different tiers of workers this may not be the case.

Do we need to look at the actual worker capacity? It seems strange that we're making a call to the `autoScaleConfig` API when it may not be used because when `workers` is empty. Perhaps the order of operations needs to change

Reply:
Refactored this. Moved all the logic for calculating total worker capacity (with and without auto scaling) to a API in the overlord. For determining the total worker capacity with auto scaling, this new API basically delegate the calculation to the configured provisioning strategy. The provisioning strategy already has a method to calculate the per worker capacity, hence, we just reuse that. Note 1: that this currently only support PendingTaskBasedWorkerProvisioningStrategy and any other provisioning strategy will result in -1 for `maximumCapacityWithAutoScale`. Note 2: PendingTaskBasedWorkerProvisioningStrategy assumes all nodes have the same capacity. This is something that already existed and is how PendingTaskBasedWorkerProvisioningStrategy determine how many new indexer/MM to create for auto scaling.

Reply:
Thanks for this refactor, it's a lot cleaner and easier to follow!